common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java

```diff
@@ -322,6 +323,7 @@ public <T> T createPayload(String action, Map<String, String> parameters) {
         if (connectorAction.isPresent() && connectorAction.get().getRequestBody() != null) {
             String payload = connectorAction.get().getRequestBody();
             payload = fillNullParameters(parameters, payload);
+            parseParameters(parameters);
```

Comment:
With this PR, can we revert this https://github.com/opensearch-project/ml-commons/pull/2811/files ?

Reply:
sure, we can do that. I think this is a more appropriate fix. Will raise a new PR to revert it.

Reply:
I will raise the new PR after this PR get merged, because it will cause conflict.

Reply:
reverting @mingshl
[Revert "Support list in response body (](https://github.com/opensearch-project/ml-commons/pull/2871/commits/247b955b5a014adf36dafb713cf6be15f8d53fcb)https://github.com/opensearch-project/ml-commons/pull/2811[)"](https://github.com/opensearch-project/ml-commons/pull/2871/commits/247b955b5a014adf36dafb713cf6be15f8d53fcb) in this PR